source/options.tsx

```diff
@@ -10,6 +10,59 @@ import * as indentTextarea from 'indent-textarea';
 
 import {perDomainOptions} from './options-storage';
 
+function setValidationStatus(text: string): void {
+	select('#validation')!.textContent = text;
+	select('form ul')!.classList.toggle('invalid', !text.includes('✔️'));
+}
+
+async function getHeaders(personalToken: string): Promise<string> {
+	const tokenLink = select<HTMLAnchorElement>('#personal-token-link')!;
+	const url = tokenLink.host === 'github.com' ?
+		'https://api.github.com/' :
+		`${tokenLink.origin}/api/v3/`;
+
+	const response = await fetch(url, {
+		cache: 'no-store',
+		headers: {
+			'User-Agent': 'Refined GitHub',
+			Accept: 'application/vnd.github.v3+json',
+			Authorization: `token ${personalToken}`
+		}
+	});
+
+	if (response.ok) {
+		setValidationStatus('✔️ Validated');
+	} else {
+		const statusText = response.status === 404 ?
+			'Invalid Domain' :
```

Comment:
Not sure if we need this. I tested on a random site so I added this check.

Reply:
Nah, it can be part of a catch-all error handler:

```ts
} else {
	try {
		setValidationStatus('❌ ' + (await response.json()).message);
	} catch (error: unknown) {
		setValidationStatus('❌ ' + (error as Error).message);
	}
}